src/python/pants/backend/codegen/protobuf/java/dependency_inference.py

```diff
@@ -39,18 +44,21 @@ async def resolve_protobuf_java_runtime_for_resolve(
     request: ProtobufJavaRuntimeForResolveRequest,
 ) -> ProtobufJavaRuntimeForResolve:
 
-    for tgt in jvm_artifact_targets:
-        if tgt[JvmResolveField].normalized_value(jvm) != request.resolve_name:
-            continue
-
-        artifact = ArtifactRequirement.from_jvm_artifact_target(tgt)
-        if (
-            artifact.coordinate.group == _PROTOBUF_JAVA_RUNTIME_GROUP
-            and artifact.coordinate.artifact == _PROTOBUF_JAVA_RUNTIME_ARTIFACT
-        ):
-            return ProtobufJavaRuntimeForResolve(tgt.address)
-
-    raise MissingProtobufJavaRuntimeInResolveError(request.resolve_name)
+    try:
+        addresses = find_jvm_artifacts_or_raise(
+            required_coordinates=[
+                UnversionedCoordinate(
+                    group=_PROTOBUF_JAVA_RUNTIME_GROUP,
+                    artifact=_PROTOBUF_JAVA_RUNTIME_ARTIFACT,
+                )
+            ],
+            resolve=request.resolve_name,
+            jvm_artifact_targets=jvm_artifact_targets,
+            jvm=jvm,
+        )
+        return ProtobufJavaRuntimeForResolve(next(iter(addresses)))
```

Comment:
I ended up switching types similar to `ProtobufJavaRuntimeForResolve` to take a collection of `Address` so as to not rely on assuming `find_jvm_artifacts_or_raise` will return one `Address` per coordinate provided. See https://github.com/pantsbuild/pants/pull/15110.

Reply:
checks out